### PR TITLE
Create archetype utils package

### PR DESCRIPTION
**Pull request by annacv**
EX-6930

### CODE REVIEW

File: packages/x-archetype-utils/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# x-archetype-utils
```
Comment (herrardo) [inline packages/x-archetype-utils/README.md:1]:
```suggestion
# x-archetype-utils

`x-archetype-utils` is a library that provides utilities to ease development. It's used in the [Interface X Archetype](https://github.com/empathyco/x-archetype) project. 

## Contributing

To start contributing to the project, please take a look at our [Contributing Guide](https://github.com/empathyco/x/blob/main/.github/CONTRIBUTING.md).
```

**Review by herrardo — CHANGES_REQUESTED** [review]

File: packages/x-archetype-utils/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  moduleFileExtensions: ['ts', 'js'],
+  transform: {
```
Comment (tiborux) [inline packages/x-archetype-utils/jest.config.js:3]: Is there a real difference between using the `preset: 'ts-jest'` or using transform for just *.ts files?

Should we change the `testEnvironment` to `jsdom`?

File: packages/x-archetype-utils/src/__tests__/sample.spec.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { x } from '../index';
+
+it('should ', function () {
```
Comment (tiborux) [inline packages/x-archetype-utils/src/__tests__/sample.spec.ts:3]:
```suggestion
it('should ', () => {
```

File: packages/x-archetype-utils/jest.config.js (hunk 2 above)
Comment (tiborux) [reply packages/x-archetype-utils/jest.config.js:3]: As javi said, use the preset

File: packages/x-archetype-utils/package.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    "url": "https://github.com/empathyco/x.git",
+    "directory": "packages/x-archetype-utils"
+  },
+  "engines": {
+    "node": ">=14.14.0"
+  },
+  "scripts": {
+    "prebuild": "rimraf dist ./*.tgz",
+    "build": "concurrently \"npm run build:*\"",
+    "build:cjs": "node_modules/.bin/tsc --project tsconfig.cjs.json",
```
Comment (tiborux) [inline packages/x-archetype-utils/package.json:30]: Remove node_modules part in all builds

```suggestion
    "build:cjs": "tsc --project tsconfig.cjs.json",
```

**Comment by tiborux** [comment]
Just a comment, you should wait until you have 2 reviewers that already post comments to apply the changes. Sometimes some of the comments can be wrong or can be discussed between the reviewers and the owner of the pr

**Review by tiborux — DISMISSED** [review]

File: packages/x-archetype-utils/package.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  "engines": {
+    "node": ">=14.14.0"
+  },
+  "scripts": {
+    "prebuild": "rimraf dist ./*.tgz",
+    "build": "concurrently \"npm run build:*\"",
+    "build:cjs": "tsc --project tsconfig.cjs.json",
+    "build:esm": "tsc --project tsconfig.esm.json",
+    "postbuild": "npm pack",
+    "test": "node_modules/.bin/jest",
```
Comment (diegopf) [inline packages/x-archetype-utils/package.json:33]:
```suggestion
    "test": "jest",
```

**Review by diegopf — CHANGES_REQUESTED** [review]

**Review by diegopf — APPROVED** [review]

**Review by herrardo — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**